File: solutions/sliding-window-maximum/sliding-window-maximum.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */

    // get window range
    // track result 

    // loop while end of window is less than nums length
        // slice window from array from start to end
        // get max from slice and push into result arr
        // increment start and end of window

var maxSlidingWindow = function(nums, k) {
    let start = 0;
    let end = start + k;
    let results = [];
        
    while (end <= nums.length) {
        let window = nums.slice(start, end);
        results.push(Math.max(...window));
        start++;
        end++;
    }
    
    return results;
};

File: solutions/sliding-window-maximum/sliding-window-maximum.py
#     get window range
#     track result 

#     loop while end of window is less than nums length
#         slice window from array from start to end
#         get max from slice and push into result arr
#         increment start and end of window

class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        start = 0
        end = start + k
        results = []
        
        while end <= len(nums):
            results.append(max(nums[start:end]))
            start += 1
            end += 1
        
        return results
